metals/src/main/scala/scala/meta/internal/rename/RenameProvider.scala

```diff
@@ -0,0 +1,275 @@
+          txtParams,
+          !occurence.symbol.desc.isType
+        )
+        refResult <- currentReferences +: (implReferences ++ companionRefs)
+        loc <- refResult.locations
+      } yield loc
+
+      val isApply =
+        symbolOccurence.exists(occ => occ._1.symbol.desc.name.value == "apply")
+
+      val allChanges = for {
+        (uri, locs) <- allReferences.toList.distinct.groupBy(_.getUri())
+      } yield {
+        val textEdits = for (loc <- locs) yield {
+          textEdit(isApply, loc, params.getNewName())
+        }
+        Seq(uri -> textEdits.toList)
+      }
+      val fileChanges = allChanges.flatten.toMap
+
+      val (openedEdits, closedEdits) = fileChanges.partition {
+        case (file, edits) =>
+          val path = AbsolutePath(Paths.get(new URI(file)))
+          buffers.contains(path)
+      }
+
+      changeClosedFiles(closedEdits)
+
+      new WorkspaceEdit(
+        openedEdits.map {
+          case (file, edits) => file -> edits.asJava
+        }.asJava
+      )
+    }
+  }
+
+  private def companionReferences(sym: String): List[ReferencesResult] = {
+    val results = for {
+      companionSymbol <- companion(sym).toIterable
+      loc <- definitionProvider.fromSymbol(companionSymbol).asScala
+    } yield referenceProvider.references(toReferenceParams(loc))
+    results.toList
+  }
+
+  private def companion(sym: String) = {
+    val termOrType = sym.desc match {
+      case Descriptor.Type(name) =>
+        Some(Descriptor.Term(name))
+      case Descriptor.Term(name) =>
+        Some(Descriptor.Type(name))
+      case other =>
+        None
+    }
+
+    termOrType.map(
+      name =>
+        Symbols.Global(
+          sym.owner,
+          name
+        )
+    )
+  }
+
+  private def changeClosedFiles(fileEdits: Map[String, List[TextEdit]]) = {
+    fileEdits.toArray.par.foreach {
+      case (file, changes) =>
+        val path = AbsolutePath(Paths.get(new URI(file)))
+        val text = path.readText
+        val newText = TextEdits.applyEdits(text, changes)
+        path.writeText(newText)
+    }
+  }
+
+  private def implementations(
+      textParams: TextDocumentPositionParams,
+      shouldCheckImplementation: Boolean
+  ) = {
+    if (shouldCheckImplementation) {
+      for (loc <- implementationProvider.implementations(textParams))
+        yield {
+          val locParams = toReferenceParams(loc)
+          referenceProvider.references(locParams)
+        }
+    } else {
+      Nil
+    }
+  }
+
+  private def canRenameSymbol(symbol: String, newName: String) = {
+    val forbiddenMethods = Set("equals", "hashCode", "unapply", "unary_!", "!")
+    val desc = symbol.desc
+    val name = desc.name.value
+    val isForbidden = forbiddenMethods(name)
+    if (isForbidden) {
+      client.logMessage(forbiddenRename(name, newName))
+    }
+    val colonNotAllowed = name.endsWith(":") && !newName.endsWith(":")
+    if (colonNotAllowed) {
+      forbiddenColonRename(name, newName)
```

Comment:
Calling this message has no side-effect
```suggestion
      client.logMessage(forbiddenColonRename(name, newName))
```

Reply:
Fixed